controller/volume_controller.go

```diff
@@ -3459,6 +3466,40 @@ func (vc *VolumeController) enqueueVolumesForBackingImageDataSource(obj interfac
 	}
 }
 
+func (vc *VolumeController) enqueueNodeChange(obj interface{}) {
+	node, ok := obj.(*longhorn.Node)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("received unexpected obj: %#v", obj))
+			return
+		}
+
+		// use the last known state, to enqueue, dependent objects
+		node, ok = deletedState.Obj.(*longhorn.Node)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("DeletedFinalStateUnknown contained invalid object: %#v", deletedState.Obj))
+			return
+		}
+	}
+
+	replicas, err := vc.ds.ListReplicasRO()
```

Comment:
You can have 2 list functions: One is for the unscheduled replicas, another is for the failed replicas **on this node only**.
This is just a tiny suggestion, I am fine with the current implementation as well.

Reply:
Thanks @shuo-wu ! Good idea as usual! 

I would like to keep the current implementation to verify the theory that the slow enqueueNodeChange function shouldn't affect the scalability